app/lib/identity_tijuana/member_sync.rb

```diff
@@ -366,6 +366,13 @@ def self.sync(user, member, sync_type)
       # puts "########## member_hash = #{member_hash.inspect}"
       # puts "########## tj_changes = #{tj_changes.inspect}"
 
+      # Check for no-op
+      if member_hash.empty? && tj_changes.empty?
+        member_info = "ID member #{member.id}" if member&.id
+        user_info = "TJ user #{user.id}" if user&.id
+        Rails.logger.warn("No changes detected for sync id: #{sync_id}, type: #{sync_type} - #{[member_info, user_info].compact.join(', ')}")
```

Comment:
This needs to be an error, doesn't it?

Reply:
Not really, the operation didn't error. We are just flagging it as it was effectively a no-op, it didn’t perform any action, nor did it impact the system's state.

It is also a likely occurrence due to sync implementation – it is possible that a member or user will be picked up for `sync` even when their attributes have not changed.

Reply:
That's an error though - the process was asked to sync this user/member, and it didn't, and so it failed. This is quite possibly where the issue we're currently sitting lies, and so we want to be notified about it, yes?

Reply:
> the process was asked to sync this user/member, and it didn't

I assumed that this sort of idempotency was intentional due to the way that `changes` are inferred from the timestamp modification, which could be a false positive, and thus `sync this member/user if there are any changes`. Eg. a member's name was changed then reverted, for sync it would appear as modified but both attribute diffs would be empty.

> This is quite possibly where the issue we're currently sitting lies, and so we want to be notified about it, yes?

Yes indeed, since we are not really interested in warnings, `error` it is :)